lib/task_listener.js

```diff
@@ -229,7 +227,18 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
-  addRunningTask(runningState) {
+  async recordCapacity () {
+    this.runtime.stats.record('capacityOverTime', {
+      duration: Date.now() - this.timestamp,
+      availableCapacity: await this.availableCapacity(),
```

Comment:
I'm not sure this is good... `availableCapacity()` has side effects...

Perhaps we should just use the latest value returned from `availableCapacity()`?
@gregarndt, do you have any suggestions here...